test/functional/page_objects/management/saved_objects_page.ts

```diff
@@ -182,53 +181,51 @@ export function SavedObjectsPageProvider({ getService, getPageObjects }: FtrProv
-        }
-        if (actionsHTML.includes('savedObjectsTableAction-relationships')) {
-          relationshipsElement = await row.findByTestSubject(
-            'savedObjectsTableAction-relationships'
-          );
-        }
-        if (actionsHTML.includes('savedObjectsTableAction-copy_saved_objects_to_space')) {
-          copySaveObjectsElement = await row.findByTestSubject(
-            'savedObjectsTableAction-copy_saved_objects_to_space'
-          );
-        }
-        return {
-          checkbox,
-          objectType: await objectType.getAttribute('aria-label'),
-          titleElement,
-          title: await titleElement.getVisibleText(),
-          menuElement,
-          inspectElement,
-          relationshipsElement,
-          copySaveObjectsElement,
-        };
-      });
+      return await Promise.all(
+        rows.map(async (row) => {
+          const checkbox = await row.findByCssSelector('[data-test-subj*="checkboxSelectRow"]');
+          // return the object type aria-label="index patterns"
+          const objectType = await row.findByTestSubject('objectType');
+          const titleElement = await row.findByTestSubject('savedObjectsTableRowTitle');
+          // not all rows have inspect button - Advanced Settings objects don't
+          // Advanced Settings has 2 actions,
+          //   data-test-subj="savedObjectsTableAction-relationships"
+          //   data-test-subj="savedObjectsTableAction-copy_saved_objects_to_space"
+          // Some other objects have the ...
+          //   data-test-subj="euiCollapsedItemActionsButton"
+          // Maybe some objects still have the inspect element visible?
+          // !!! Also note that since we don't have spaces on OSS, the actions for the same object can be different depending on OSS or not
+          const actions = await row.findByClassName('euiTableRowCell--hasActions');
+          // getting the innerHTML and checking if it 'includes' a string is faster than a timeout looking for each element
+          const actionsHTML = await actions.getAttribute('innerHTML');
+          const [
+            menuElement = null,
+            inspectElement = null,
+            relationshipsElement = null,
+            copySaveObjectsElement = null,
+          ] = await Promise.all([
+            actionsHTML.includes('euiCollapsedItemActionsButton') &&
+              row.findByTestSubject('euiCollapsedItemActionsButton'),
+            actionsHTML.includes('savedObjectsTableAction-inspect') &&
+              row.findByTestSubject('savedObjectsTableAction-inspect'),
+            actionsHTML.includes('savedObjectsTableAction-relationships') &&
+              row.findByTestSubject('savedObjectsTableAction-relationships'),
+            actionsHTML.includes('savedObjectsTableAction-copy_saved_objects_to_space') &&
+              row.findByTestSubject('savedObjectsTableAction-copy_saved_objects_to_space'),
```

Comment:
Calling the 4 `findByTestSubject` async calls concurrnetly instead of blocking on each call.
Outside the scope of the PR but since I was refactoring the function anyways i thought i'd change it.